### PR TITLE
feat: Bitquery module.

### PR DESCRIPTION
**Pull request by ixm-759**

### CODE REVIEW

File: src/stats/stats.service.ts [hunk 1]
```diff
@@ -576,40 +581,32 @@ export class StatsService {
...
-    const decimals = await bananaContract.methods.decimals().call();
-
-    const [burned, supply] = await Promise.all([
-      bananaContract.methods.balanceOf(burnAddress()).call(),
-      bananaContract.methods.totalSupply().call(),
-    ]);
-
-    const burntAmount = burned / 10 ** decimals;
-    const totalSupply = supply / 10 ** decimals;
-    const circulatingSupply = totalSupply - burntAmount;
```
Comment (ape-swap) [inline src/stats/stats.service.ts:590]: Lets keep using RPC whenever its possible and makes sense. It is more real time and precise.

File: src/stats/stats.service.ts [hunk 2]
```diff
@@ -576,40 +581,32 @@ export class StatsService {
...
-    const decimals = await gnanaContract.methods.decimals().call();
-
-    const [treasury, supply] = await Promise.all([
-      gnanaContract.methods.balanceOf(gBananaTreasury()).call(),
-      gnanaContract.methods.totalSupply().call(),
-    ]);
-
-    const treasuryAmount = treasury / 10 ** decimals;
-    const totalSupply = supply / 10 ** decimals;
-    const circulatingSupply = totalSupply - treasuryAmount;
```
Comment (ape-swap) [inline src/stats/stats.service.ts:611]: Lets keep using RPC whenever its possible and makes sense. It is more real time and precise.

**Review by ape-swap — CHANGES_REQUESTED** [review]
Very good work!  Left a few comments on the code. 

Also please remember to add `Dto` as a sufix of the new Dtos to follow that nomenclature.

File: src/bitquery/bitquery.controller.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @Param('network') network: string,
+  ): Promise<TokenInformation> {
+    this.logger.debug(`Called GET /token/${network}/${address}`);
+    return await this.bitqueryService.getTokenInformation(address, network);
+  }
+  @Get('/candle/:address')
+  async getCandleToken(
+    @Param('address') address: string,
+    @Query() candleOptionsDto: CandleOptions,
+  ): Promise<any> {
```
[the remark this answers is not in the document]
Comment (ixm-759) [reply src/bitquery/bitquery.controller.ts:46]: Done!

File: src/bitquery/bitquery.service.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+  async getCandleToken(address: string, candleOptions: CandleOptions) {
+    const network = 'bsc';
+    const { from, to } = candleOptions;
+    let { minTrade, interval } = candleOptions;
+
+    let query;
+    try {
+      interval = interval || 60;
+      minTrade = minTrade || 0;
```
[the remark this answers is not in the document]
Comment (ixm-759) [reply src/bitquery/bitquery.service.ts]: Done!

File: src/bitquery/dto/candle.dto.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export class CandleOptions {
```
[the remark this answers is not in the document]
Comment (ixm-759) [reply src/bitquery/dto/candle.dto.ts]: Done!

File: src/bitquery/dto/pairInformation.dto.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+}
+export class PairInformation {
+  ticker_id?: string;
+  addressLP?: string;
+  base?: Token;
+  target?: Token;
+  liquidity?: number;
+  quote?: QuoteToken;
+  createdAt?: string;
+}
```
[the remark this answers is not in the document]
Comment (ixm-759) [reply src/bitquery/dto/pairInformation.dto.ts]: Done!

File: src/stats/stats.service.ts (hunk 1 above)
Comment (ixm-759) [reply src/stats/stats.service.ts]: Rollback! Done!

File: src/stats/stats.service.ts (hunk 2 above)
Comment (ixm-759) [reply src/stats/stats.service.ts]: Rollback! Done!

**Pull request merged**